webapp/channels/src/components/admin_console/schema_admin_settings.tsx

```diff
@@ -37,23 +42,53 @@ import Setting from './setting';
 
 import './schema_admin_settings.scss';
 
-export default class SchemaAdminSettings extends React.PureComponent {
-    static propTypes = {
-        config: PropTypes.object,
-        environmentConfig: PropTypes.object,
-        setNavigationBlocked: PropTypes.func,
-        schema: PropTypes.object,
-        roles: PropTypes.object,
-        license: PropTypes.object,
-        editRole: PropTypes.func,
-        updateConfig: PropTypes.func.isRequired,
-        isDisabled: PropTypes.bool,
-        consoleAccess: PropTypes.object,
-        cloud: PropTypes.object,
-        isCurrentUserSystemAdmin: PropTypes.bool,
-    };
+type ConsoleAccess = {
+    read: Record<string, boolean>;
+    write: Record<string, boolean>;
+}
 
-    constructor(props) {
+type Props = {
+    schema: any;
+    setNavigationBlocked?: (blocked: boolean) => void;
+    isDisabled?: boolean;
+    consoleAccess?: ConsoleAccess;
+    enterpriseReady?: boolean;
+    license?: Record<string, string>;
+    config?: DeepPartial<AdminConfig>;
+    clientConfig?: ClientConfig;
+    environmentConfig?: Partial<EnvironmentConfig>;
+    roles?: Record<string, Role>;
+    editRole?: (role: Role) => void;
+    updateConfig?: any;
+    cloud?: CloudState;
+    isCurrentUserSystemAdmin?: boolean;
+}
+
+type Settings = {
+    [key: string]: any;
+}
+type State = {
+    saveNeeded: boolean | string;
+    saving: boolean;
+    serverError: null | string;
+    errorTooltip: boolean;
+    customComponentWrapperClass: string;
+    confirmNeededId: string;
+    showConfirmId: string;
+    clientWarning: string;
+    prevSchemaId?: string;
+    [key: string]: any;
```

Comment:
This effectively removes the benefit we get from typing here, because `State` now just allows for anything to be defined. We shouldn't need this.

Reply:
@devinbinnie because of ```schema: any``` , i am setting ```[key: string]: any;```, if we define schema's type, I can reuse it here

Reply:
What errors do we get by removing this? I think we need to be a bit more prescriptive here.